src/components/tabs/descendants.tsx

```diff
@@ -0,0 +1,422 @@
+/*
```

Comment:
Descendant's related functions and custom hooks.

Descendants are all the elements within a tree that can be controlled using a state manager.